src/main/java/edu/whimc/observationdisplayer/quests/QuestObservationObjective.java

```diff
@@ -0,0 +1,38 @@
+package edu.whimc.observationdisplayer.quests;
+
+import edu.whimc.observationdisplayer.events.ObserveEvent;
+import me.blackvein.quests.CustomObjective;
+import me.blackvein.quests.Quest;
+import me.blackvein.quests.Quests;
+import org.bukkit.Bukkit;
+import org.bukkit.event.EventHandler;
+import org.bukkit.plugin.Plugin;
+
+/**
+ * A custom observation objective using the Quest plugin.
+ */
+public class QuestObservationObjective extends CustomObjective {
+
+    public QuestObservationObjective() {
+        this.setName("Observation Objective");
+        this.setAuthor("Pieter Svenson");
+        this.setShowCount(true);
+        this.setCountPrompt("Enter the number of observations the player must make:");
+        this.setDisplay("Make %count% observations");
+    }
+
+    @EventHandler
+    public void onObserve(ObserveEvent event) {
+
+        Plugin plugin = Bukkit.getServer().getPluginManager().getPlugin("Quests");
+        if (plugin == null) {
+            return;
+        }
+        Quests quests = (Quests) plugin;
```

Comment:
Is this necessary to do every time? You can do it like in [the tutorial](https://github.com/PikaMug/Quests/wiki/Master-%E2%80%90-Custom-Quest-API). Since Quests takes care of registering the event, the plugin should exist if the event even gets called.

Reply:
Additionally, you should add quests as a soft depend in the `plugin.yml`.

Reply:
I generally like getting fresh instances where possible to reduce field clutter within classes. Given that these events are thrown very infrequently, I figure it's not a big deal and it's worth saving the headache of a possible NPE which Bukkit plugins are notorious for.